sigar_linux_common.go

```diff
@@ -181,44 +180,58 @@ func (self *ProcList) Get() error {
 }
 
 func (self *ProcState) Get(pid int) error {
-	contents, err := readProcFile(pid, "stat")
+	data, err := readProcFile(pid, "stat")
 	if err != nil {
 		return err
 	}
 
-	headerAndStats := strings.SplitAfterN(string(contents), ")", 2)
-	pidAndName := headerAndStats[0]
-	fields := strings.Fields(headerAndStats[1])
-
-	name := strings.SplitAfterN(pidAndName, " ", 2)[1]
-	if name[0] == '(' && name[len(name)-1] == ')' {
-		self.Name = name[1 : len(name)-1] // strip ()'s
-	} else {
-		return errors.New(fmt.Sprintf("Malformed process stats for pid %d", pid))
+	// Extract the comm value with is surrounded by parentheses.
+	lIdx := bytes.Index(data, []byte("("))
+	rIdx := bytes.LastIndex(data, []byte(")"))
+	if lIdx < 0 || rIdx < 0 {
```

Comment:
Nit: Should return an error also when `lIdx >= rIdx || rIdx + 2 >= len(data)`
to avoid crashing below.